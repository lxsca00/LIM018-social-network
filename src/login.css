#container-login {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    border-radius: 15px;
    margin-right: 5%;
    margin-left: 5%;
    padding: 5%;
    /*background-color: rgb(129, 120, 120);*/
}

.image-login {
    justify-content: center;
    border-radius: 80%;
    background-color: rgb(250, 244, 244);
    width: 40%;
    height: 40%;
    /*background-color: rgb(196, 202, 202);*/ 
}

.h2-login{
    display: flex;
    justify-content: center;
    text-align: center;
    font-size: 180%;
}

.form-login{
    display: flex;
    justify-content: center;
    width: 90%;
    height: 90%;
    /*background-color: rgb(248, 149, 20);*/
    margin-right: 5%;
    margin-left: 5%;
}

.h1-login{
    margin-top: 5%;
    margin-bottom: 5%;
    text-align: center;
    font-size: 200%;
}

.p-login {
    text-align: center;
    font-size: 10px; 
}

.logo-container{
    display: flex;
    justify-content: center;
    gap: 2rem;
}

.image-google{
    border-radius: 70%;
    width: 10%;
    height: 10%;
    
}
.image-facebook{
    border-radius: 60%;
    width: 10%;
    height: 10%;   
}

@media screen and (min-width: 500px) {

    .container-login{
        display: inline-block;
        
    }
    .image-login{
        border-radius: 50%;
        float: left;
        width: 20%;
        height: 20%;
    }
    .h2-login{
        font-size: 180%;
    }

    .form-login{
        display: flex;
        justify-content: center;
        width: 50%;
        height: 50%;
        /*background-color: rgb(248, 149, 20);*/

    }
    .image-google{
        flex-direction: column;
        border-radius: 70%;
        width: 2%;
        height: 2%;
        
    }
    .image-facebook{
        flex-direction: column;
        border-radius: 60%;
        width: 2%;
        height: 2%;   
    }
}