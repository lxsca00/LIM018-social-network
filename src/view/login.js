//  Vista de la página de ingreso

export default () => {
  /*document.getElementById("containerNav").style.visibility = "hidden";*/
  const loginView = `
    <div class = 'container-login'>
    <div class="h2-login">
      <h2 class="text-center"> ¡Bienvenidos a PopcornZone! </h2>
    </div>
      <img class="image-login" src="https://i.pinimg.com/originals/54/cc/e0/54cce0449cfd4414fdc19b068a97e00a.png">

        <form id = "form-login" class="form-login"> 
          <h1 class="h1-login">Inicia Sesión</h1>
          <input type="email" placeholder="Correo Electrónico" id="ingresaEmail" class="input-login-email" >
          <input type="password"  placeholder="Contraseña" id="ingresaContrasena"class="input-login-password" >
          <button type="submit" class="btnIngresar" id="btnIngresar"> Ingresar </button>
          <p class ="p-login"> -------------O ingresa con: -----------------------</p>
          <button type="submit" class= "btnGoogle id="iniciaGoogle">Ingresa con Google</button>
          <button type="submit" class="btnRegistrar" id="btnRegistrar">¿No tienes cuenta? Regístrate</button>
        </form>
        
   </div>`;
   
  

   const divLogin = document.createElement('div');
   divLogin.id = 'login';
   divLogin.innerHTML = loginView;
   return divLogin;
 };



/*<img class="image-facebook" src="https://i.pinimg.com/736x/8e/fb/55/8efb55e9efc12eb11bedf41caa7f33bb.jpg">
          <img class="image-google" src="https://w7.pngwing.com/pngs/543/934/png-transparent-google-app-logo-google-logo-g-suite-google-text-logo-circle.png">
        */