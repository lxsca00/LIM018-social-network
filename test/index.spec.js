/**
 * @jest-environment jsdom
 */

import login from '../src/view/login.js';
// import { fEventLogin } from '../src/funcionesDom';
import { eventLogin } from '../src/lib/index.js';

jest.mock('../src/lib/index.js');

// login
describe('login', () => {
  it('debería ser una función', () => {
    expect(typeof login).toBe('function');
  });
  it('click para login', () => {
    document.body.appendChild(login());
    document.getElementById('login-email').value = 'sss@gmail.com';
    document.getElementById('login-password').value = '123456';
    const btnLogin = document.getElementById('login-button');
    expect(btnLogin instanceof HTMLElement).toBe(true);
    // console.log(fEventLogin());
    console.log(eventLogin());
    expect(document.body).toMatchSnapshot();
  });
});
