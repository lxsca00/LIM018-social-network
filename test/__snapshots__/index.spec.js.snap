// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`login click para login 1`] = `
<body>
  <div
    id="login"
  >
    
  
    <div
      id="container-login"
    >
      
    
      <div
        class="fondo-logo"
      >
        
      
        <img
          class="imagen-logo"
          src="https://i.pinimg.com/originals/54/cc/e0/54cce0449cfd4414fdc19b068a97e00a.png"
        />
        
    
      </div>
      
    
      <form
        class="form-login"
        id="form-login"
      >
         
      
        <h1>
          Inicia sesión
        </h1>
        
      
        <p
          id="login-message"
        >
           Por favor inicia sesión para continuar 
        </p>
        
      
        <input
          id="login-email"
          placeholder="Correo Electrónico"
          type="email"
        />
        
      
        <input
          id="login-password"
          placeholder="Contraseña"
          type="password"
        />
        
      
        <button
          class="btnInicio"
          id="login-button"
          type="submit"
        >
           INGRESAR 
        </button>
        
      
        <p
          class="p-login"
        >
           O ingresa con: 
        </p>
        
      
        <div
          class="logo-container"
        >
          
      
          <img
            class="image-facebook"
            src="https://i.pinimg.com/736x/8e/fb/55/8efb55e9efc12eb11bedf41caa7f33bb.jpg"
          />
          
      
          <img
            class="image-google"
            src="https://w7.pngwing.com/pngs/543/934/png-transparent-google-app-logo-google-logo-g-suite-google-text-logo-circle.png"
          />
          
    
        </div>
        
    
        <p>
           ¿No tienes cuenta? 
          <a
            href="#/registro"
          >
             Regístrate 
          </a>
           
        </p>
        
    
      </form>
      
    
      <div
        class="background-modal-error"
      >
        
      
        <div
          class="modal-error"
        >
          
        
          <p
            class="login-error"
          >
             Hay un error 
          </p>
          
        
          <button
            class="close-modal btnInicio"
          >
             CERRAR 
          </button>
          
      
        </div>
        
    
        <div>
          
  
        </div>
      </div>
    </div>
  </div>
</body>
`;
